test/splitgraph/commands/test_multicorn_fdws.py

```diff
@@ -0,0 +1,985 @@
+        }
+    elif data_source == "pg":
+        # Ensure query is going to be aggregated on the foreign server
+        assert _extract_pg_queries_from_explain(result)[0] == (
+            'SELECT count(*) AS "count.*" '
+            "FROM public.account "
+            "WHERE public.account.firstname NOT LIKE 'Su_an%%'"
+        )
+
+    # Ensure results are correct
+    result = test_local_engine.run_sql(query, return_shape=ResultShape.ONE_ONE)
+    assert result == 996
+
+
+@pytest.mark.mounting
+@pytest.mark.parametrize("data_source", ["es", "pg"])
+@pytest.mark.usefixtures("esorigin_fdw")
+@pytest.mark.usefixtures("pgorigin_sqlalchemy_fdw")
+def test_simple_aggregation_functions_filtering(data_source, test_local_engine):
+    query = f"""
+    SELECT avg(age), max(balance)
+    FROM {data_source}.account
+    WHERE balance > 20000 AND age < 30
+    """
+
+    # Ensure query is going to be aggregated on the foreign server
+    result = test_local_engine.run_sql("EXPLAIN " + query)
+
+    if data_source == "es":
+        assert _extract_es_queries_from_explain(result)[0] == {
+            "query": {
+                "bool": {
+                    "must": [
+                        {"range": {"balance": {"gt": "20000"}}},
+                        {"range": {"age": {"lt": 30}}},
+                    ]
+                }
+            },
+            "aggs": {
+                "avg.age": {"avg": {"field": "age"}},
+                "max.balance": {"max": {"field": "balance"}},
+            },
+        }
+    elif data_source == "pg":
+        assert _extract_pg_queries_from_explain(result)[0] == (
+            'SELECT avg(public.account.age) AS "avg.age", max(public.account.balance) AS "max.balance" '
+            "FROM public.account "
+            "WHERE public.account.balance > 20000 AND public.account.age < 30"
+        )
+
+    # Ensure results are correct
+    result = test_local_engine.run_sql(query)
+    assert len(result) == 1
+
+    # Assert aggregation result
+    assert math.isclose(result[0][0], 24.4399, rel_tol=1e-05)
+    assert result[0][1] == 49795.0
+
+    # Variant with COUNT(*)
+    query = f"SELECT avg(balance), COUNT(*) FROM {data_source}.account WHERE firstname ~~ 'Su_an%'"
+
+    # Ensure query is going to be aggregated on the foreign server
+    result = test_local_engine.run_sql("EXPLAIN " + query)
+
+    if data_source == "es":
+        assert _extract_es_queries_from_explain(result)[0] == {
+            "query": {"bool": {"must": [{"wildcard": {"firstname": "Su?an*"}}]}},
+            "track_total_hits": True,
+            "aggs": {
+                "avg.balance": {"avg": {"field": "balance"}},
+            },
+        }
+    elif data_source == "pg":
+        assert _extract_pg_queries_from_explain(result)[0] == (
+            'SELECT avg(public.account.balance) AS "avg.balance", count(*) AS "count.*" '
+            "FROM public.account WHERE public.account.firstname LIKE 'Su_an%%'"
```

Comment:
The original query has `WHERE firstname ~~ 'Su_an%'` whereas this one has ` WHERE public.account.firstname LIKE 'Su_an%%'` -- shouldn't it also only have one percentage sign?

Reply:
As discussed offline, the answer is no, and the reason is documeted here: https://docs.sqlalchemy.org/en/14/faq/sqlexpressions.html#why-are-percent-signs-being-doubled-up-when-stringifying-sql-statements